File: Ejercicios POM/POMSAUCEDEMO/pageobjectmodel/src/main/java/pages/SauceDemoProductsPage.java
package pages;


import java.util.List;
import java.util.Map;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import actions.Action;
import io.cucumber.datatable.DataTable;
import net.serenitybdd.core.pages.PageObject;
import net.thucydides.core.annotations.DefaultUrl;
import utils.Constants;

@DefaultUrl("https://www.saucedemo.com/inventory.html")
public class SauceDemoProductsPage extends PageObject {

	By selectFilter = By.xpath("//*[@class='product_sort_container']");
	By listProductsNames = By.xpath("//*[@class='inventory_list']/descendant::div[@class='inventory_item_name']");
	By listProductsPrices = By.xpath("//*[@class='inventory_list']//descendant::div[@class='inventory_item_price']");
	

	public void navegar(String url) {

		Action.navegateToUrl(getDriver(), Constants.getData(url));
	}

	//----------------Filtros----------------------
	//Metodo para filtro de A la Z
	public void filtroAtoZ(DataTable dataTable) {
		//Mapeo
		Map<String, String> data = dataTable.asMap(String.class, String.class);
		Action.selectFromOption(getDriver(), data.get("filtrado"), selectFilter);
	}

	public void filtroZtoA(DataTable dataTable) {

		Map<String, String> data = dataTable.asMap(String.class, String.class);
		Action.selectFromOption(getDriver(), data.get("filtro"), selectFilter);

	}
	
	public void filtroBajo(DataTable dataTable) {

		Map<String, String> data = dataTable.asMap(String.class, String.class);
		Action.selectFromOption(getDriver(), data.get("filtroBajo"), selectFilter);

	}

	public void filtroAlto(DataTable dataTable) {
		
		Map<String, String> data = dataTable.asMap(String.class, String.class);
		Action.selectFromOption(getDriver(), data.get("filtro"), selectFilter); 
	}
	//------------------------------------------------------------------
	//----------------Impresion de informacion por consola----------------------
	private void printGeneralTitle(int opcFiltro){
		//    1    2       3         4
		//{ AtoZ, ZtoA, LowtoHigh, HighToLow}
		String title = "----- Filtro: ";
		switch(opcFiltro){
			case 1:
				title+="AtoZ";
				title+=" ---- \n----- Nombres Productos -----\n";
			break;
			case 2:
				title+="ZtoA";
				title+=" ---- \n----- Nombres Productos -----\n";
			break;
			case 3:
				title +="LowtoHigh";
				title+=" ---- \n----- Precios Productos -----\n";
			break;
			case 4:
				title +="HighToLow";
				title+=" ---- \n----- Precios Productos -----\n";
			break;
			default:
				System.out.println("Opcion invalida");
			break;
		} 
		System.out.print(title);

	}

	public void printProducts(int opcFiltro) {
		//    1    2       3         4
		//{ AtoZ, ZtoA, LowtoHigh, HighToLow}
		List<WebElement> list;
		printGeneralTitle(opcFiltro);
		if(opcFiltro == 1 || opcFiltro ==2 )
		{
			list=Action.obtainElementsList(getDriver(),listProductsNames);

		}else
		{
			list=Action.obtainElementsList(getDriver(),listProductsPrices);
		}
		
		for (WebElement product : list) {
			System.out.println(product.getText());
		}
		
	}
	//---------------------------------------------------------------
	
	

}

File: Ejercicios PLAYWRIGHT/PLAYWRIGHTSAUCEDEMO/src/pages/SauceDemoProductsPage.ts
import {expect, Locator, Page} from '@playwright/test'

export class SauceDemoProductsPage{

    readonly page:Page;
    private selectFilter: Locator;
    private listProductsPrices: string[];
    private listProductsNames: string[] ;
    readonly pointerProductsNames: Locator;
    readonly pointerProductsPrices: Locator;
    

    constructor(page:Page)
    {
        this.page = page;
        this.selectFilter = page.locator('xpath=//*[@class="product_sort_container"]');
        this.pointerProductsNames = page.locator('xpath=//*[@class="inventory_list"]//descendant::div[@class="inventory_item_name"]');
        this.pointerProductsPrices = page.locator('xpath=//*[@class="inventory_list"]//descendant::div[@class="inventory_item_price"]');
    }
    
    

    async filtroAtoZ()
    {
        // Selecciona el filtro por valor
        await this.selectFilter.selectOption('az'); 
        // Toma toda la lista del nombre de los productos
        this.listProductsNames = await this.pointerProductsNames.allInnerTexts();
        await this.page.waitForTimeout(3000);
    }

    async filtroZtoA()
    { 
        await this.selectFilter.selectOption('za'); 
        this.listProductsNames = await this.pointerProductsNames.allInnerTexts();
        await this.page.waitForTimeout(3000);
        
    }

    async filtroAlto()
    {
       await this.selectFilter.selectOption('hilo'); 
       this.listProductsPrices = await this.pointerProductsPrices.allInnerTexts();
       await this.page.waitForTimeout(3000);
    }
    async filtroBajo()
    {
        await this.selectFilter.selectOption('lohi'); 
        this.listProductsPrices = await this.pointerProductsPrices.allInnerTexts();
        await this.page.waitForTimeout(3000);
        
    }

    printProductsPrices(title)
    {
        
        console.log(title);
       for (let i= 0; i < this.listProductsPrices.length; i++) {
            console.log(this.listProductsPrices[i]);
        
       }
    }

    printProductsNames(title)
    {
        
        console.log(title);
        
       for (let i= 0; i < this.listProductsNames.length; i++) {
            console.log(this.listProductsNames[i]);
        
       }
    }

    
}